src/http/request_handler.rs

```diff
@@ -80,7 +80,45 @@ pub async fn graphql_request<T: DeserializeOwned + GraphQLRequestLike>(
     }
   }
 }
+async fn url_req<T: DeserializeOwned + GraphQLRequestLike>(
+  req: Request<Body>,
+  server_ctx: &ServerContext,
+) -> Result<Response<Body>> {
+  let req_ctx = Arc::new(create_request_context(&req, server_ctx));
+  if let Some(query) = req.uri().path_and_query() {
+    let query = query.as_str();
+    let mut parser = Parser::from_path(query)?;
+    let request = parser.parse::<T>();
+    match request {
+      Ok(request) => {
+        let response = request.data(req_ctx.clone()).execute(&server_ctx.schema).await;
+        let mut response = update_cache_control_header(response, server_ctx, req_ctx).to_response()?;
+        update_response_headers(&mut response, server_ctx);
+        Ok(response)
+      }
+      Err(err) => {
+        log::error!("Failed to parse request: {query}",);
+
+        let mut response = async_graphql::Response::default();
+        let server_error = ServerError::new(format!("Unexpected GraphQL Request: {}", err), None);
+        response.errors = vec![server_error];
 
+        Ok(GraphQLResponse::from(response).to_response()?)
```

Comment:
There are different kinds of errors and since this is a RESTish API, we should stick to those standards. Eg: if the operation name is not provided a `404` should be issued. If the query params can't be parsed `400` should be issued etc.